packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -146,6 +165,78 @@ function rebaseMark<TNodeChange>(
             return clone(currMark);
         }
         default:
-            fail("Not implemented");
+            fail(`Unsupported mark type: ${baseType}`);
+    }
+}
+
+function handleCurrAttach<T>(
+    currMark: Attach<T>,
+    factory: MarkListFactory<T>,
+    lineageRequests: LineageRequest<T>[],
+    offset: number,
+) {
+    const rebasedMark = clone(currMark);
+    factory.pushContent(rebasedMark);
+    lineageRequests.push({ mark: rebasedMark, offset });
+}
+
+function isAttachAfterBaseAttach<T>(currMark: Attach<T>, baseMark: Attach<T>): boolean {
+    const lineageCmp = compareLineages(currMark.lineage, baseMark.lineage);
+    if (lineageCmp < 0) {
+        return false;
+    } else if (lineageCmp > 0) {
+        return true;
+    }
+
+    // TODO: Handle tiebreak
+    return false;
+}
+
+function compareLineages(
+    lineage1: LineageEvent[] | undefined,
+    lineage2: LineageEvent[] | undefined,
+): number {
+    if (lineage1 === undefined || lineage2 === undefined) {
+        return 0;
```

Comment:
This doesn't seem to allow for cases where we wan to return "they need to be tie-breaked by according to the base mark's tiebreak".
Here's the example scenario we discussed before:
Staring state: a b
A1) Delete a b
A2) Insert c
B) Insert x between a and b

Reply:
Also, it would be nice to add a test for this scenario.

Reply:
This function returns zero when the lineage does not determine the mark order. In that case we can use the tiebreak policy in `isAttachAfterBaseAttach`.

Reply:
I agree there's a code-structure question here. My point is that if this function just returns 0 in the scenario above then it's on `isAttachAfterBaseAttach` to go dig in the lineage to figure out whether the tiebreak of the base insert or the tiebreak from the curr insert should be obeyed. I'm open to either code-structure.

Reply:
Ah, I forgot our earlier conclusion about the correct behavior of this scenario. Thanks for reminding me.

Reply:
I don't think it requires a significant restructuring of the code, but I would prefer to implement this in a separate PR.

Reply:
Sounds good. Please add a TODO comment in case we don't get to it immediately.